### PR TITLE
added check to pandas splitter

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

Add checker to make sure that the ratios sum to 1

### Checklist:

- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [x] I have added tests.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by yueguoguo** [comment]
Note the ratio does not have to be summed to 1. If they are not, the codes will do the normalization. This is done by the function `process_split_ratio` in [here](https://github.com/Microsoft/Recommenders/blob/master/reco_utils/dataset/split_utils.py).

File: reco_utils/dataset/split_utils.py [hunk 1]
```diff
@@ -152,6 +152,9 @@ def split_pandas_data_with_ratios(data, ratios, seed=1234, resample=False):
     Returns:
         list: List of pd.DataFrame splitted by the given specifications.
     """
+    if sum(ratios) != 1.0:
+        raise ValueError("The ratios have to sum to 1")
```
Comment (anargyri) [inline reco_utils/dataset/split_utils.py]: Could you also add the requirement "The ratios have to sum to 1" in the docstring.

File: tests/unit/test_python_splitter.py [hunk 2]
```diff
@@ -79,6 +79,30 @@ def random_date_generator(start_date, range_in_days):
     return rating
 
 
+@pytest.fixture(scope="module")
+def python_int_dataset(test_specs):
+    # fix the the random seed
```
Comment (anargyri) [inline tests/unit/test_python_splitter.py]: remove extra "the"

File: tests/unit/test_python_splitter.py [hunk 3]
```diff
@@ -79,6 +79,30 @@ def random_date_generator(start_date, range_in_days):
...
+    return np.random.randint(
+        low=0,
+        high=6,
+        size=(test_specs["number_of_users"], test_specs["number_of_items"]),
+    )
+
+
+@pytest.fixture(scope="module")
+def python_float_dataset(test_specs):
+    # fix the the random seed
```
Comment (anargyri) [inline tests/unit/test_python_splitter.py]: here too

File: tests/unit/test_python_splitter.py [hunk 4]
```diff
@@ -79,6 +79,30 @@ def random_date_generator(start_date, range_in_days):
...
+    np.random.seed(test_specs["seed"])
+
+    # generates the user/item affinity matrix. Ratings are from 1 to 5, with 0s denoting unrated items
```
Comment (anargyri) [inline tests/unit/test_python_splitter.py]: maybe better say that ratings are in the interval [0, 5)

File: reco_utils/dataset/split_utils.py (hunk 1 above)
Comment (miguelgfierro) [reply reco_utils/dataset/split_utils.py]: good suggestion

**Pull request merged**